### PR TITLE
Move scripts from old website to CI 

### PR DESCRIPTION
**Pull request by aliabd**
This PR creates a new workflow in the ci that: 

- uploads a wheel of the library to aws on every commit to main 
- uploads all the website demos to spaces on every commit to main (with `_main` suffix) 
- uploads all the website demos to spaces when a new version is released (no `_main` suffix) 

I tested the workflow on a fork of this repo and finally got it working fine [here](https://github.com/aliabd/gradio/actions/runs/5910947339/job/16033015853), and all the spaces were updated correctly. 

Also makes some related changes: 
- Adds `LoginButton`, `LogoutButton`, and `DuplicateButton` component demos, and fixes issues with other component demos.
- Fixes model3d guide and broken demos inside it. 
- Removes an old workflow that used to create the docs when a new gradio version is released on the old website.

Closes: #5204 #5211 #5255

### CODE REVIEW

File: .github/workflows/upload-website-demos.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
+name: Upload Website Demos to Spaces
+
+on:
+  workflow_dispatch:
+  pull_request:
+    types: closed
+    branches:
+      - main
```
Comment (pngwn) [inline .github/workflows/upload-website-demos.yml:8]: Could we just do this on `push` to `main` instead?

File: .github/workflows/upload-website-demos.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+jobs:
+  upload-wheel-aws:
+    if: github.event.pull_request.merged == true || github.event.action == ''
```
Comment (pngwn) [inline .github/workflows/upload-website-demos.yml:11]: We could get rid of this with the above change.

File: .github/workflows/upload-website-demos.yml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    name: Upload Demos
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.ref }}
+          repository: ${{ github.event.pull_request.head.repo.full_name }}
```
Comment (pngwn) [inline .github/workflows/upload-website-demos.yml:19]: And this.

**Review by pngwn — COMMENTED** [review]
Can we reuse the existing `upload_demo_to_spaces` script and extend the build/deploy PR workflow (adding a `push: branches: -main` event and changing the demos based on the event)? These changes are mostly duplicates of those files.

**Comment by freddyaboulton** [comment]
> Can we reuse the existing upload_demo_to_spaces script and extend the build/deploy PR workflow build/deploy PR workflow 

Makes sense!

**Comment by aliabd** [comment]
@pngwn @freddyaboulton I got this working, removed the new workflow and instead worked it into build-pr/deploy-pr-to-spaces. I did keep the script though as it does a bunch of stuff differently from upload_demo_to_space. 

I tested the on a fork here: 
- Example of [build-pr](https://github.com/aliabd/gradio/actions/runs/6017964902/job/16325287238) and [deploy-pr-to-spaces](https://github.com/aliabd/gradio/actions/runs/6017992326/job/16325366948) working fine in this [PR](https://github.com/aliabd/gradio/pull/9). You can see [spaces](https://huggingface.co/spaces/gradio-pr-deploys/pr-9-all-demos) used the correct [wheel](https://huggingface.co/spaces/gradio-pr-deploys/pr-9-all-demos/blob/cc463830e4528e4a703145241ca28af1943b83e0/requirements.txt#L3) representing the right [commit](https://github.com/aliabd/gradio/commit/ebf397f0ef9a54330d3422d8fb84935c03f9b496) inside the PR. 
- Example of [build-pr](https://github.com/aliabd/gradio/actions/runs/6018101315/job/16325685718) and [deploy-pr-to-spaces](https://github.com/aliabd/gradio/actions/runs/6018119373/job/16325731373) working fine when the same PR got merged. You can see all the website demos used the correct [wheel](https://huggingface.co/spaces/gradio/hello_world_main/blob/53b4e8ae8e688ed23a602abdaf1d7538a3334e6c/requirements.txt#L1) representing the right [commit](https://github.com/aliabd/gradio/commit/d4396fb6e412e57adfa8956a891e92fd2cc93092) to main.  

(Note there are small differences between the code here and on the fork: basically was easier to test with a different aws bucket I control so tried it there with a different s3 url, but will use @freddyaboulton's bucket here)

**Review by freddyaboulton — APPROVED** [review]
Thanks @aliabd ! Looks good to me!

**Pull request merged**